### ISSUE TITLE
[FR] Official MKS TS35 v2 support 

### ISSUE THREAD

**Issue by kitor** [thisiskeithb/PrusaAIO#45]
### Is your feature request related to a problem? Please describe.

TL;DR: Some time ago I bought MKS Robin Nano in set with TS35 v2 display. I'm running `prusa-aio-bugfix-2.1.x`, it works perfectly after just a few lines modified, in all 3 UI variants. I think it can be officially supported, unless there's some blocker I'm not aware of.

### Are you looking for hardware support?

MKS TS35 v2 LCD display.

### Describe the feature you want

To have official `PRUSA_AIO_DISPLAY_MKS_TS35` support in Prusa AIO :)

### Additional context

Looking at my `git diff`, what I defined is below, including ugly hack to get PrusaAIO not to complain :)

I tested all three UI types (`TFT_CLASSIC_UI`, `TFT_COLOR_UI` and `TFT_LVGL_UI`), they seem to work fine (including their known limitations) - I'm using `TFT_COLOR_UI` now.

At least v2 display doesn't have SD card slot so it has to be defined as onboard one.

I was unsure how to properly incorporate this into a feature and push a PR, thus hope this will help to see what I've done.

`#define ENCODER_STEPS_PER_MENU_ITEM 2` was necessary at least in my case, without that it was often missing steps. 

```diff
diff --git a/Marlin/Configuration.h b/Marlin/Configuration.h
index cd8c428efa..b4088a8f2a 100644
--- a/Marlin/Configuration.h
+++ b/Marlin/Configuration.h
@@ -2721,7 +2721,7 @@
 // Use this option to override the number of step signals required to
 // move between next/prev menu items.
 //
-//#define ENCODER_STEPS_PER_MENU_ITEM 1
+#define ENCODER_STEPS_PER_MENU_ITEM 2
 
 /**
  * Encoder Direction Options
@@ -3292,7 +3292,7 @@
 // Usually paired with MKS Robin Nano V2 & V3
 // https://github.com/makerbase-mks/MKS-TFT-Hardware/tree/master/MKS%20TS35
 //
-//#define MKS_TS35_V2_0
+#define MKS_TS35_V2_0
 
 //
 // 320x240, 2.4", FSMC Display From MKS
@@ -3396,7 +3396,7 @@
  *   root of your SD card, together with the compiled firmware.
  */
 //#define TFT_CLASSIC_UI
-//#define TFT_COLOR_UI
+#define TFT_COLOR_UI
 //#define TFT_LVGL_UI
 
 #if ENABLED(TFT_COLOR_UI)
@@ -3453,7 +3453,7 @@
 //
 // Touch Screen Settings
 //
-//#define TOUCH_SCREEN
+#define TOUCH_SCREEN
 #if ENABLED(TOUCH_SCREEN)
   #define BUTTON_DELAY_EDIT      50 // (ms) Button repeat delay for edit screens
   #define BUTTON_DELAY_MENU     250 // (ms) Button repeat delay for menus
```

```diff
diff --git a/Marlin/Prusa_AIO_Configuration.h b/Marlin/Prusa_AIO_Configuration.h
index 811b79a854..324401384e 100644
--- a/Marlin/Prusa_AIO_Configuration.h
+++ b/Marlin/Prusa_AIO_Configuration.h
@@ -410,12 +410,13 @@
  *
  * LCD Covers: https://github.com/thisiskeithb/PrusaAIO/wiki/Recommended-LCD-Covers
  */
-#define PRUSA_AIO_DISPLAY_STOCK_REPRAP_DISCOUNT_SMART_CONTROLLER
+//#define PRUSA_AIO_DISPLAY_STOCK_REPRAP_DISCOUNT_SMART_CONTROLLER
 //#define PRUSA_AIO_DISPLAY_OLED_REPRAP_DISCOUNT_SMART_CONTROLLER
 //#define PRUSA_AIO_DISPLAY_BTT_DUAL_MODE_TFT
 //#define PRUSA_AIO_DISPLAY_BTT_MINI_12864_NEOPIXEL
 //#define PRUSA_AIO_DISPLAY_FYSETC_MINI_12864_NEOPIXEL
 //#define PRUSA_AIO_DISPLAY_MKS_MINI_12864_NEOPIXEL
+#define PRUSA_AIO_DISPLAY_MKS_TS35
 
 //
 // Reverse the encoder direction everywhere.
```

```diff
diff --git a/Marlin/src/inc/Prusa_AIO_Conditionals.h b/Marlin/src/inc/Prusa_AIO_Conditionals.h
index 129cfaa96c..c8c6922b1f 100644
--- a/Marlin/src/inc/Prusa_AIO_Conditionals.h
+++ b/Marlin/src/inc/Prusa_AIO_Conditionals.h
@@ -158,7 +158,7 @@
   #define PRUSA_AIO_HAS_CONTROLLER_FAN 1
   #define CONTROLLER_FAN_PIN HEATER_1_PIN                         // PB0
   #define PRUSA_AIO_HAS_ENDSTOP_INTERRUPTS_FEATURE 1
-  #define PRUSA_AIO_SDCARD_CONNECTION LCD
+  #define PRUSA_AIO_SDCARD_CONNECTION ONBOARD
   #define PRUSA_AIO_NEEDS_FAN_SOFT_PWM 1
   #define PRUSA_AIO_PROBE_USES_Z_MIN_ENDSTOP_PIN
   #define PRUSA_AIO_USB_SERIAL_PORT -1
```

```diff
diff --git a/Marlin/src/inc/Prusa_AIO_SanityCheck.h b/Marlin/src/inc/Prusa_AIO_SanityCheck.h
index 50cb3df127..da948eeac2 100644
--- a/Marlin/src/inc/Prusa_AIO_SanityCheck.h
+++ b/Marlin/src/inc/Prusa_AIO_SanityCheck.h
@@ -198,9 +198,9 @@
 //========================= LCD / TFT / Controller ==========================
 //===========================================================================
 
-#if MANY(PRUSA_AIO_DISPLAY_STOCK_REPRAP_DISCOUNT_SMART_CONTROLLER, PRUSA_AIO_DISPLAY_OLED_REPRAP_DISCOUNT_SMART_CONTROLLER, PRUSA_AIO_DISPLAY_BTT_DUAL_MODE_TFT, PRUSA_AIO_DISPLAY_BTT_MINI_12864_NEOPIXEL, PRUSA_AIO_DISPLAY_FYSETC_MINI_12864_NEOPIXEL, PRUSA_AIO_DISPLAY_MKS_MINI_12864_NEOPIXEL, PRUSA_AIO_DISPLAY_BTT_TFT35_SPI_V1, PRUSA_AIO_DISPLAY_MKS_TS35_V2_0)
+#if MANY(PRUSA_AIO_DISPLAY_STOCK_REPRAP_DISCOUNT_SMART_CONTROLLER, PRUSA_AIO_DISPLAY_OLED_REPRAP_DISCOUNT_SMART_CONTROLLER, PRUSA_AIO_DISPLAY_BTT_DUAL_MODE_TFT, PRUSA_AIO_DISPLAY_BTT_MINI_12864_NEOPIXEL, PRUSA_AIO_DISPLAY_FYSETC_MINI_12864_NEOPIXEL, PRUSA_AIO_DISPLAY_MKS_MINI_12864_NEOPIXEL, PRUSA_AIO_DISPLAY_BTT_TFT35_SPI_V1, PRUSA_AIO_DISPLAY_MKS_TS35_V2_0, PRUSA_AIO_DISPLAY_MKS_TS35)
```

**Comment by thisiskeithb** [comment]
Duplicate of https://github.com/thisiskeithb/PrusaAIO/discussions/30
